src/sentry/api/endpoints/organization_integration_request.py

```diff
@@ -0,0 +1,72 @@
+from __future__ import absolute_import
+
+from django.core.urlresolvers import reverse
+
+from rest_framework.response import Response
+
+from sentry import options
+from sentry.api.bases.organization_integrations import (
+    OrganizationIntegrationBaseEndpoint,
+    OrganizationIntegrationsPermission,
+)
+from sentry.models import Integration, OrganizationMember
+from sentry.utils.email import MessageBuilder
+from sentry.utils.http import absolute_uri
+
+
+class OrganizationIntegrationRequestEndpoint(OrganizationIntegrationBaseEndpoint):
+    permission_classes = (OrganizationIntegrationsPermission,)
+
+    def post(self, request, organization, integration_id):
+        """
+        Email the default organization owner asking to install an integration.
+        ``````````````````````````````````````````````````````````````````````
+
+        TODO 0.0 descibe why this exists.
+
+        :param string message: Optional message from the requester to the owner.
+        """
+
+        # If for some reason the user had permissions all along, silently fail.
+        requester = request.user
+        if requester.id in map(lambda user: user.id, organization.get_owners()):
+            return Response({"detail": "User can install integration"}, status=200)
+
+        # In the edge case where an admin adds the integration between the user
+        # seeing and clicking the button, just silently fail.
+        installed_integration = self.get_organization_integration(organization, integration_id)
+        if installed_integration and installed_integration.status == ObjectStatus.ACTIVE:
+            return Response({"detail": "Integration already installed"}, status=200)
+
+        # TODO 1.0 is this the right way? Is there a more explicit primary key call?
+        integration = Integration.objects.filter(integration_id=integration_id)
```

Comment:
@mgaeta I am confused why’d  this even works since `integration_id1` is not a field on that model. I think you are getting confused with `OrganizationIntegration` which points to a specific integration.